frontend/markdown-pdf/build.js

```diff
@@ -113,77 +157,668 @@ Promise.all([
+                        var config = {};
+                        if (frontmatter.data.config) {
+                            frontmatter.data.config.forEach(elem => {
+                                config[elem.name] = elem.content;
+                            });
+                        } else {
+                            // If there is no config, this md file does not need to be converted to a booklet
+                            return;
+                        }
+
+                        // If we don't need to generate a booklet for this md file, skip this file
+                        if (!config.generate_booklet) {
+                            return;
+                        }
+                    } else {
+                        // If this md file does not exist in the root folder, we can skip it.
+                        return;
+                    }
+
+                    console.log(`Generating booklet for ${manual}`);
+
+                    // Create a booklet for each language set in the frontmatter
+                    return Promise.all(
+                        Array.from(Object.keys(config.languages)).map(
+                            async set => {
+                                // Create an array with the different languages + ordering for this booklet
+                                const langs = ordering.filter(value => {
+                                    return (
+                                        languages.get(manual).has(value) &&
+                                        config.languages[set].includes(value)
+                                    );
+                                });
+
+                                // Print a warning to console.error when a language is specified in frontmatter, but no html is generated for that language.
+                                config.languages[set].forEach(lang => {
+                                    if (!languages.get(manual).has(lang)) {
+                                        console.error(
+                                            `Language ${lang} is specified in front matter of ${manual}, but cannot find markdown file ${lang}/${manual}`
+                                        );
+                                    }
+                                });
+
+                                // Generate the front page of this booklet
+                                const frontPage = generateFrontPagePDF(
+                                    browser,
+                                    manual,
+                                    css,
+                                    fpCSS,
+                                    logoSVG,
+                                    langs,
+                                    /* Dynamically generate link to online documentation */
+                                    DOCS_BASE_URL +
+                                        manual.replace('.md', '.html')
```

Comment:
Kunnen we de extensie alleen vervangen als deze aan 't eind van de filename zit? Zie regex die op andere plaatsen gebruikt wordt. Ook hoofdletterongevoeligheid gaat issues voorkomen.